src/Sylius/Bundle/ApiBundle/Resources/config/api_resources/OrderPayment.xml

```diff
@@ -0,0 +1,31 @@
+<?xml version="1.0" ?>
+
+<!--
+
+ This file is part of the Sylius package.
+
+ (c) Paweł Jędrzejewski
+
+ For the full copyright and license information, please view the LICENSE
+ file that was distributed with this source code.
+
+-->
+
+<resources xmlns="https://api-platform.com/schema/metadata"
+           xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+           xsi:schemaLocation="https://api-platform.com/schema/metadata https://api-platform.com/schema/metadata/metadata-2.0.xsd"
+>
+    <resource class="%sylius.model.order.class%" shortName="OrderPayment">
```

Comment:
sth is wrong with this config 🗡️  it is named as `OrderPayment`, class is an `order` short name is `OrderPayment`, and this `GET` function returns `paymentConfiguration` :D
Definitely, we need to disuse it on audio because I am not sure what do you want to achieve here :D

Reply:
yes, thats true. I wasnt sure tho how to handle it proper way so i left it with my first concept (working one) as there's some problems like unability to get payment resource as guest :/

Reply:
Maybe you should think about rout like `/new-api/shop/orders/{id}/payments/{paymentId}/configuration` ? We have similar functionality in `/new-api/shop/orders/{id}/payments/{paymentId}` and guest can get payment. Its resolved in `Sylius\Bundle\ApiBundle\Command\Checkout\ChoosePaymentMethod` and his handler. If you would take this solution you can remove this file and custom controller next all this logic can be handle in command and command handler